### PR TITLE
Detect variables that are not plain strings

### PR DESCRIPTION
**Pull request by ajwootto**
Add logic for detecting when a variable is not a plain string, and output such cases as "notices" for the user to add an alias

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -70,7 +70,7 @@
     "postpack": "shx rm -f oclif.manifest.json",
     "posttest": "yarn lint",
     "prepack": "yarn build && oclif manifest && oclif readme",
-    "test": "mocha --forbid-only \"test/**/*.test.ts\"",
+    "test": "mocha --forbid-only \"test/**/nodejs.test.ts\"",
```
Comment (laurawarr) [inline package.json:73]: I'm assuming this is intended to be temporary?

Comment (laurawarr) [reply package.json:73]: you can run `yarn test -g nodejs`

Comment (ajwootto) [reply package.json]: nice I didn't know that

File: src/utils/diff/parsers/common.ts [hunk 2]
```diff
@@ -149,17 +184,30 @@ export abstract class BaseParser {
...
+    match(content: string): MatchResult | null {
+        const pattern = this.buildRegexPattern()
+
+        const matches = pattern.exec(content)
+        if (matches) {
+            const varName = matches[1] || matches[2]
+            return {
+                isUnknown: !varName.match(/^["'].*["']$/),
+                // position or named parameter match
+                name: varName.replace(/["']/g, ''),
```
Comment (laurawarr) [inline src/utils/diff/parsers/common.ts:196]:
```suggestion
                name: varName.replace(/^['"]|['"]$/g, ''),
```
what about something like this? to only replace quotes at the start or end of the string

File: src/utils/diff/parsers/common.ts [hunk 3]
```diff
@@ -149,17 +184,30 @@ export abstract class BaseParser {
...
+        const pattern = this.buildRegexPattern()
+
+        const matches = pattern.exec(content)
+        if (matches) {
+            const varName = matches[1] || matches[2]
+            return {
+                isUnknown: !varName.match(/^["'].*["']$/),
+                // position or named parameter match
+                name: varName.replace(/^["']|["']$/g, ''),
+                matches: matches
```
Comment (laurawarr) [inline src/utils/diff/parsers/common.ts:197]: we could just add the content and index directly to this object, rather than the entire RegExpExecArray

**Review by laurawarr — APPROVED** [review]
looks like there's one warning about an unused `AxiosError` in `api/variables.ts` that could be cleaned up

**Pull request merged**